src/main/java/com/google/sps/servlets/HandlePhotoShoppingServlet.java

```diff
@@ -109,12 +112,20 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
    * Returns the shopping query by calling methods from the photo content detection classes, 
    * based on the {@code photoCategory}, passing {@code uploadedImageBytes} as argument.
    */
-  private String getQuery(String photoCategory, byte[] uploadedImageBytes) throws IllegalArgumentException {
+  private String getQuery(String photoCategory, byte[] uploadedImageBytes)
+      throws IOException, PhotoDetectionException {    
     switch (photoCategory) {
       case "product":
         return "Fountain pen";
       case "shopping-list":
-        return "Fuzzy socks";
+        ImageTextDectector imageTextDectector = new ImageTextDectector();
+        String shoppingQuery;
+        try {
+          shoppingQuery = imageTextDectector.imageToShoppingListExtractor(uploadedImageBytes);
+        } catch (PhotoDetectionException | IOException exception) {
```

Comment:
I'm thinking now, is it necessary to catch IOException? I think this can be avoided - I looked in the ImageTextDectector.java and I saw that on lines 76-77 IOException is caught and PhotoDetectionException thrown, which is also what I do and I think is so clean! Is any other code in ImageTextDectector that may cause an IOException which is not caught (I'm thinking not, but not sure)? If not, then IOException can be removed from the exceptions thrown in  ImageTextDectector methods and also, from here. If yes, how about catching it and re-throwing PhotoDetectionException?